pkg/ingester/ingester.go

```diff
@@ -791,7 +816,8 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 		case err != nil && errors.Is(err, errEphemeralStorageDisabledForUser):
 			// Add all samples for ephemeral series as "failed".
 			for _, ts := range req.EphemeralTimeseries {
-				ephemeralStats.failedSamplesCount += len(ts.Samples)
+				ephemeralStats.failedSamplesCount += len(ts.Samples) + len(ts.Histograms)
```

Comment:
This change looks a bit weird to me. What if we just track samples to `failedSamplesCount` but in places where we read it and we want to sum histograms too, we also sum there `failedHistogramsCount`? Reason is that just reading at this code looks a "bug" we both track failed histograms to `failedSamplesCount` and `failedHistogramsCount`.

Reply:
Keeping the current logic, an example of misleading info is in the span log `"event", "start commit"`, where histogram failures are tracked both as samples and histograms.

Reply:
We had a version where that was the case, but the diff was huge and also it would mean that you need to rename failedSamplesCount to failedFloatSamplesCount or something like that, since in general we're trying to mean all kinds of samples (float and histogram) when saying sample. This is a compromise to be able to get some details in the logs, but keep the diff small.

Reply:
If we want another rework of this, I suggest to do it in a separate PR